jaeger-core/src/main/java/com/uber/jaeger/samplers/PerOperationSampler.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2016, Uber Technologies, Inc
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.uber.jaeger.samplers;
+
+import com.uber.jaeger.samplers.http.OperationSamplingParameters;
+import com.uber.jaeger.samplers.http.PerOperationSamplingParameters;
+
+import java.util.HashMap;
+import lombok.AccessLevel;
+import lombok.Data;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Computes {@link #sample(String, long)} using the name of the operation, and maintains a specific
+ * {@link GuaranteedThroughputSampler} instance for each operation.
+ */
+@Slf4j
+@Data
+@Getter(AccessLevel.NONE)
+public class PerOperationSampler implements Sampler {
+  private final HashMap<String, GuaranteedThroughputSampler> operationNameToSampler;
+  private final int maxOperations;
+  private ProbabilisticSampler defaultSampler;
+  private double lowerBound;
+  private double defaultSamplingProbability;
```

Comment:
I think we should have this discussion separately, I don't like it because when used that way, the GuaranteedTrhoughputSampler no longer fulfills it's guarantee.